source/common/quic/envoy_quic_server_session.cc

```diff
@@ -180,5 +180,15 @@ void EnvoyQuicServerSession::storeConnectionMapPosition(FilterChainToConnectionM
   position_.emplace(connection_map, filter_chain, position);
 }
 
+quic::QuicSSLConfig EnvoyQuicServerSession::GetSSLConfig() const {
+  quic::QuicSSLConfig config = quic::QuicServerSessionBase::GetSSLConfig();
+  config.early_data_enabled = position_.has_value()
+                                  ? static_cast<const QuicServerTransportSocketFactory&>(
```

Comment:
The other place in the code-base I see this cast it is a dynamic_cast, in `EnvoyQuicProofSource::getTlsCertConfigAndFilterChain`. Should this be the same?